### PR TITLE
feat(EMS-3344): no PDF - application submission - XLSX - buyer

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:

This PR updates the XLSX mapping sent during application submission so that the "Buyer" section is aligned with the latest no PDF/XLSX design:

### E2E directories
- Split up application submission tests into new directories, for faster GHA execution and improved readability/maintainability:
  - `/application-submission/business-conditions/*.spec.js`
  - `/application-submission/buyer-conditions/*.spec.js`
  - `/application-submission/export-contract-conditions/*.spec.js`
  - `/application-submission/policy-conditions/*.spec.js`

### E2E commands
- Create new DRY cypress commands to click an "alternative currency" radio option and submit an alternative currency:
  - `completeAlternativeCurrencyField`
  - `clickAlternativeCurrencyRadioAndCompleteCurrency`
  - `clickAlternativeCurrencyRadioAndSubmitCurrency
- Update `completeBuyerSection` and various other cypress commands to accept new params and conditionally submit an alternative buyer currency.

### E2E test coverage improvements
- Add new application submission E2E tests for single and multiple policies for the following scenarios: 
  - Buyer fully populated with all possible conditions.
  - Exporter has a connection with buyer.
  - Exporter has traded with buyer, with outstanding payments, failed to pay on time.
  - Exporter has traded with buyer, with outstanding payments, not failed to pay on time.
  - Exporter has traded with buyer, no outstanding payments, failed to pay on time.
  - Exporter has  traded with buyer, no outstanding payments, not failed to pay on time.
  - Exporter is connected with buyer and has traded with the buyer before.
  - Exporter has had credit insurance cover with the buyer before.
  - Exporter has buyer financial information.
  - Buyer "total outstanding payments" and "amount overdue" with an alternative currency.
  - Buyer "financial information” as no.

### API alignments, helpers

- Align `CURRENCY` and `YOUR_BUYER` field ID constants.
- Align `YOUR_BUYER` content strings.
- Add missing `VALIDATION` constants.
- Add missing `FORM_TITLES` content strings.
- Update the `mapYesNoField` helper function to:
  - Have object structured params.
  - Accept and return a custom `defaultValue`.
- Update `getPopulatedApplication` helper function return a fully populated buyer (buyer relationship, buyer trading history).

### API XLSX generation

- Bump the `DECLARATIONS` index in `TITLE_INDEXES`.
- Update `XLSX_ROW_INDEXES` to bump some particular indexes accordingly, depending on the conditional buyer answers.
- Update XLSX specific content strings.
- Update `mapBuyer` XLSX function and create new mapping functions to generate new answers in no PDF:
  - `mapConnectionWithBuyer`
  - `mapBuyerTradingHistory`
  - `mapOutstandingPayments`
  - `mapPreviousCoverWithBuyer`
- Update unit tests.

## Miscellaneous :heavy_plus_sign:

- Fix some E2E linting issues.
- Minor E2E fixture improvements.
- Fix some typos.
- Create new DRY application fixture, `mockApplicationMinimalBrokerBuyerAndCompany`.
- Fix/update/align API buyer types.
- Update `test.yml`.
- Fix a failing date test.
- Update some `getPopulatedApplication` unit tests to use `toThrow()`

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/application-submission/multiple-policy-type/policy-conditions/submit-an-application-multiple-policy-type-maximum-cover.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { APPLICATION } from '../../../../../../../constants';
+
+context(
+  'Insurance - submit an application - Multiple policy type, no broker - As an Exporter, I want to submit my completed credit insurance application with a value greater than 500000, So that UKEF can process and make a decision on my application',
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/application-submission/multiple-policy-type/policy-conditions/submit-an-application-multiple-policy-type-maximum-cover.spec.js:4]: should we add without-broker to name of file?

File: e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/submit-an-application-single-policy-type-buyer-no-financial-accounts.spec.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
-context('Insurance - submit an application - Single policy type, no broker - As an Exporter, I want to submit my completed credit insurance application with a value greater than 500000, So that UKEF can process and make a decision on my application', () => {
+context('Insurance - submit an application - Single policy type, fully populated buyer', () => {
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/submit-an-application-single-policy-type-buyer-no-financial-accounts.spec.js:1]: add no financial accounts to context?

File: e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/traded-with-buyer/outstanding-payments-alternative-currency.spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+context('Insurance - submit an application - Single policy type, exporter has traded with buyer, has outstanding payments, alternative currency', () => {
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/traded-with-buyer/outstanding-payments-alternative-currency.spec.js:1]: add buyer failed to pay on time to context?

File: e2e-tests/insurance/cypress/e2e/journeys/application-submission/multiple-policy-type/policy-conditions/submit-an-application-multiple-policy-type-maximum-cover.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/application-submission/multiple-policy-type/policy-conditions/submit-an-application-multiple-policy-type-maximum-cover.spec.js:4]: Instead i've removed it from the test description for both single and multiple "maximum cover" tests. It doesn't make sense to say "no broker" as there are many other conditions not mentioned. Would be overkill to list them all

File: e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/submit-an-application-single-policy-type-buyer-no-financial-accounts.spec.js (hunk 2 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/application-submission/single-policy-type/buyer-conditions/submit-an-application-single-policy-type-buyer-no-financial-accounts.spec.js:1]: Good shout, updated for single and multiple policy tests

File: src/api/generate-xlsx/map-application-to-XLSX/map-buyer/map-connection-with-buyer/index.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { XLSX } from '../../../../content-strings';
+import xlsxRow from '../../helpers/xlsx-row';
+import { ApplicationBuyerRelationship } from '../../../../types';
+
+const { CONNECTION_WITH_BUYER, CONNECTION_WITH_BUYER_DESCRIPTION } = FIELD_IDS;
+
+const { FIELDS } = XLSX;
+
+/**
+ * mapPreviousCoverWithBuyer
```
Comment (Zainzzkk) [inline src/api/generate-xlsx/map-application-to-XLSX/map-buyer/map-connection-with-buyer/index.ts:11]: not a blocker, wrong function

File: src/api/helpers/get-populated-application/index.api-error.test.ts [hunk 5]
```diff
@@ -179,6 +179,28 @@ describe('api/helpers/get-populated-application - error handling', () => {
     }
   });
 
+  it('should throw an error when buyerRelationship does not exist', async () => {
+    const invalidId = applicationIds.id;
+
+    try {
+      await getPopulatedApplication(context, { ...applicationIds, buyerRelationship: invalidId });
```
Comment (Zainzzkk) [inline src/api/helpers/get-populated-application/index.api-error.test.ts:186]: could we use `.toThrow()`?

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**